ui/src/plugin/PluginDetailView.tsx

```diff
@@ -0,0 +1,259 @@
+import React, {Component} from 'react';
+import {RouteComponentProps} from 'react-router';
+import ReactMarkDown from 'react-markdown';
+import MonacoEditor from 'react-monaco-editor';
+import axios from 'axios';
+import Info from '@material-ui/icons/Info';
+import Build from '@material-ui/icons/Build';
+import Subject from '@material-ui/icons/Subject';
+import Refresh from '@material-ui/icons/Refresh';
+import Button from '@material-ui/core/Button';
+import DefaultPage from '../common/DefaultPage';
+import * as config from '../config';
+import Container from '../common/Container';
+
+interface IProps extends RouteComponentProps<{id: string}> {}
+
+interface IState {
+    pluginInfo: IPlugin | null;
+    displayText: string | null;
+    currentConfig: string | null;
+}
+
+export default class PluginDetailView extends Component<IProps, IState> {
+    public state: IState = {pluginInfo: null, displayText: null, currentConfig: null};
+
+    public componentDidMount() {
+        const {match} = this.props;
+        axios.get<IPlugin[]>(`${config.get('url')}plugin`).then((response) => {
```

Comment:
The plugins should already be loaded inside the PluginStore, could you inject the pluginStore into the PluginDetailView and then use the already existing values there? The pluginStore does also already have a getByID method which can be used here.